strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/StoragesConfigurationController.java

```diff
@@ -332,4 +347,51 @@ public ResponseEntity removeRepository(@RepositoryMapping Repository repository,
         }
     }
 
+    @ApiOperation(value = "Deletes recursively the content of path located under the defined storageId and repositoryId.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The repository path was deleted successfully."),
+                            @ApiResponse(code = 404, message = "The repository path cannot be deleted because storageId,repositoryId,or path was not found!"),
+                            @ApiResponse(code = 500, message = "Failed to remove the repository path") })
+    @PreAuthorize("hasAuthority('CONFIGURATION_DELETE_REPOSITORY_PATH')")
+    @DeleteMapping(value = "/delete", consumes = MediaType.APPLICATION_JSON_VALUE, produces = { MediaType.TEXT_PLAIN_VALUE,
+                                                                                                MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity removeRepositoryPath(@ApiParam(value = "The path object", required = true)
+                                               @RequestBody @Validated({ Default.class,
+                                                                         ProxyConfigurationFormChecks.class})
+                                               PathForm pathForm,
+                                               BindingResult bindingResult,
+                                               @RequestHeader(HttpHeaders.ACCEPT) String accept)
+    {
+
+        if(bindingResult.hasErrors())
+        {
+            return getFailedResponseEntity(HttpStatus.NOT_FOUND, FAILED_REPOSITORY_PATH_REMOVAL_FORM_ERROR, accept);
+        }
+
+        try
+        {
+            final RepositoryPath repositoryPath = repositoryPathResolver.resolve(pathForm.getStorageId(),
+                                                                                 pathForm.getRepositoryId(),
+                                                                                 pathForm.getPath());
```

Comment:
Right before this, I would add a validation to check the storage and repository existence. If they don't exist, a 404 exception with a proper message will be thrown.